numba/parfor.py

```diff
@@ -393,6 +406,66 @@ def _numpy_to_parfor(self, lhs, expr):
         # return error if we couldn't handle it (avoid rewrite infinite loop)
         raise NotImplementedError("parfor translation failed for ", expr)
 
+    def _numpy_map_to_parfor(self, call_name, lhs, args, kws, expr):
+        """generate parfor from Numpy calls that are maps.
+        """
+        scope = lhs.scope
+        loc = lhs.loc
+        arr_typ = self.typemap[lhs.name]
+        el_typ = arr_typ.dtype
+
+        # generate loopnests and size variables from lhs correlations
+        loopnests = []
+        size_vars = []
+        index_vars = []
+        for this_dim in range(arr_typ.ndim):
+            corr = self.array_analysis.array_shape_classes[lhs.name][this_dim]
+            size_var = self.array_analysis.array_size_vars[lhs.name][this_dim]
+            size_vars.append(size_var)
+            index_var = ir.Var(scope, mk_unique_var("parfor_index"), loc)
+            index_vars.append(index_var)
+            self.typemap[index_var.name] = types.int64
+            loopnests.append( LoopNest(index_var, 0, size_var, 1, corr) )
+
+        # generate init block and body
+        init_block = ir.Block(scope, loc)
+        init_block.body = mk_alloc(self.typemap, self.calltypes, lhs,
+            tuple(size_vars), el_typ, scope, loc)
+        body_label = next_label()
+        body_block = ir.Block(scope, loc)
+        expr_out_var = ir.Var(scope, mk_unique_var("$expr_out_var"), loc)
+        self.typemap[expr_out_var.name] = el_typ
+
+        index_var, index_var_typ = self._make_index_var(scope, index_vars, body_block)
+
+        if call_name=='zeros':
+            value = ir.Const(0, loc)
+        elif call_name=='ones':
+            value = ir.Const(1, loc)
+        elif call_name=='random.ranf':
+            # reuse the call expr for single value
+            expr.args = []
+            self.calltypes.pop(expr)
+            self.calltypes[expr] = self.typemap[expr.func.name].get_call_type(typing.Context(), [], {})
+            value = expr
+        else:
+            NotImplementedError("Numpy map to parfor")
```

Comment:
Changed the error message. This actual checking for supported functions is done in `_is_supported_npycall`. We have unsupported Numpy functions in our tests that would trigger this in case of introduced bugs.